pkg/defaults/defaults.go

```diff
@@ -154,17 +154,19 @@ func FromConfig(
 			addProvidesForStep(initialReleaseStep, params)
 			buildSteps = append(buildSteps, initialReleaseStep)
 
-		} else if rawStep.TestStepConfiguration != nil && rawStep.TestStepConfiguration.OpenshiftInstallerClusterTestConfiguration != nil && rawStep.TestStepConfiguration.OpenshiftInstallerClusterTestConfiguration.Upgrade {
-			var err error
-			step, err = clusterinstall.E2ETestStep(*rawStep.TestStepConfiguration.OpenshiftInstallerClusterTestConfiguration, *rawStep.TestStepConfiguration, params, podClient, templateClient, secretGetter, artifactDir, jobSpec)
-			if err != nil {
-				return nil, nil, fmt.Errorf("unable to create end to end test step: %v", err)
+		} else if testStep := rawStep.TestStepConfiguration; testStep != nil {
+			if testStep.OpenshiftInstallerClusterTestConfiguration != nil {
+				if testStep.OpenshiftInstallerClusterTestConfiguration.Upgrade {
```

Comment:
nit: those are a lot of `if`s. Can you put them in a func to be easier to read. Something like 
```go
func isStepWithClusterUpgrade(rawStep api.StepConfiguration) bool {
	testStep: = rawStep.TestStepConfiguration
	if testStep != nil && testStep.OpenshiftInstallerClusterTestConfiguration != nil {
		if testStep.OpenshiftInstallerClusterTestConfiguration.Upgrade {
			return true
		}
	}
	return false
}
```